pandas/tests/io/excel/test_openpyxl.py

```diff
@@ -130,6 +131,19 @@ def test_engine_kwargs_append_data_only(ext, data_only, expected):
             DataFrame().to_excel(writer, sheet_name="Sheet2")
 
 
+@pytest.mark.parametrize("kwarg_name", ["read_only", "data_only"])
+@pytest.mark.parametrize("kwarg_value", [True, False])
+def test_engine_kwargs_append_reader(datapath, ext, kwarg_name, kwarg_value):
+    # GH 55027
+    # test that `read_only` and `data_only` can be passed to
+    #  `openpyxl.reader.excel.load_workbook` via `engine_kwargs`
+    filename = datapath("io", "data", "excel", "test1" + ext)
+    with contextlib.closing(
+        OpenpyxlReader(filename, engine_kwargs={kwarg_name: kwarg_value})
+    ) as reader:
+        assert getattr(reader.book, kwarg_name) == kwarg_value
```

Comment:
Right, it only tests that those arguments are forwarded. That was intentional, as this was exactly what was not possible before. Passing `data_only=False` is no problem, it will just provide any formula as string instead of the computed result. I can add a test for that. I don't see how any of the current options to [openpyxl.reader.excel.load_workbook](https://openpyxl.readthedocs.io/en/stable/api/openpyxl.reader.excel.html#openpyxl.reader.excel.load_workbook) could break anything in pandas reading the data. The overall structure will remain the same only the type of some cells would be different (e.g. formula string instead of a float results).

Reply:
Thinking about this again, we do not test this with other engines. I'm good here.